### ISSUE TITLE
CliffordAlgebra Ownership and Device Synchronization in nn.Module

### ISSUE THREAD

**Issue by Concode0** [Concode0/Versor#11]
### TL;DR

The CliffordAlgebra instance is being silently registered as a submodule in classes inheriting from nn.Module (Losses/Activations), leading to redundant memory allocation, ownership conflicts, and RuntimeError during device migration (CPU/MPS/CUDA).

### Affected Area

Models, layers, and functional are all spread out. (Tangled depending on the development order)

### Algebra Signature

Cl(*,*,*) always

### What's Happening

```shell
Following the 2026-03-26 refactor where CliffordAlgebra became an nn.Module,
 any class using standard assignment (self.algebra = algebra) within its __init__ silently 
triggers PyTorch's submodule registration. This cause Memory Pressure and Device Collision.
```

### Expected Behavior

More better Perf and lower memory usage.

### Minimal Reproduction

```python

```

### Checks Already Tried (for instability / performance)

- [ ] Reduced learning rate
- [ ] Reduced bivector norm clip (`optimizer.bv_clip`)
- [ ] Switched to standard Adam
- [ ] Reduced model depth / channels
- [ ] Used `.reshape()` instead of `.view()` after multi-rotor ops
- [ ] Passed `device='cpu'` explicitly to `CliffordAlgebra`
- [ ] Tried a different `exp_policy` (`auto` / `fast` / `exact`)

### Device

- [ ] CPU
- [ ] CUDA
- [ ] Apple Silicon (MPS)

### Versor Commit / Python / PyTorch

69bc3ba, Python 3.11.12, PyTorch 2.10.0 (macOS, Darwin 25.3.0, Apple Silicon)

**Comment by Concode0** [comment]
Main engine logic seems stabilized. Remaining example tasks will be updated lazily as they don't block core functionality. Closing this only when the entire ecosystem is synchronized.

**Comment by Concode0** [comment]
Done, See f5048d3167748d6c340505cb2520d596398e4667, fc04b5a949b07a4fcc8c686d5e364f16af44ede5, a5cf0be276210f5869f8c4307b7b6b284e9a9484.
